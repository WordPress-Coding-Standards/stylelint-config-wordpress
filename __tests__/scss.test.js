'use strict';

const fs = require( 'fs' ),
	config = require( '../scss.js' ),
	stylelint = require( 'stylelint' ),
	validScss = fs.readFileSync( './__tests__/scss-valid.scss', 'utf-8' ),
	invalidScss = fs.readFileSync( './__tests__/scss-invalid.scss', 'utf-8' );

describe( 'flags no warnings with valid scss', () => {
	let result;

	beforeEach( () => {
		result = stylelint.lint( {
			code: validScss,
			config,
		} );
	} );

	it( 'did not error', () => {
		return result.then( ( data ) => expect( data.errored ).toBeFalsy() );
	} );

	it( 'flags no warnings', () => {
		return result.then( ( data ) =>
			expect( data.results[ 0 ].warnings ).toHaveLength( 0 )
		);
	} );
} );

describe( 'flags warnings with invalid scss', () => {
	let result;

	beforeEach( () => {
		result = stylelint.lint( {
			code: invalidScss,
			config,
		} );
	} );

	it( 'did error', () => {
		return result.then( ( data ) => expect( data.errored ).toBeTruthy() );
	} );

	it( 'flags correct number of warnings', () => {
		return result.then( ( data ) =>
			expect( data.results[ 0 ].warnings ).toHaveLength( 8 )
		);
	} );
} );
