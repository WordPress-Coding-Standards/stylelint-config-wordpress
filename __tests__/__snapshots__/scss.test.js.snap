// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flags warnings with invalid scss snapshot matches warnings 1`] = `
Array [
  Object {
    "column": 1,
    "line": 1,
    "rule": "scss/at-rule-no-unknown",
    "severity": "error",
    "text": "Unexpected unknown at-rule \\"@unknown\\" (at-rule-no-unknown) (scss/at-rule-no-unknown)",
  },
  Object {
    "column": 2,
    "line": 12,
    "rule": "scss/at-if-closing-brace-space-after",
    "severity": "error",
    "text": "Expected single space after \\"}\\" of @if statement (scss/at-if-closing-brace-space-after)",
  },
  Object {
    "column": 1,
    "line": 14,
    "rule": "scss/at-else-empty-line-before",
    "severity": "error",
    "text": "Unxpected empty line before @else (scss/at-else-empty-line-before)",
  },
  Object {
    "column": 2,
    "line": 12,
    "rule": "scss/at-if-closing-brace-newline-after",
    "severity": "error",
    "text": "Unexpected newline after \\"}\\" of @if statement (scss/at-if-closing-brace-newline-after)",
  },
  Object {
    "column": 2,
    "line": 22,
    "rule": "at-rule-empty-line-before",
    "severity": "error",
    "text": "Unexpected empty line before at-rule (at-rule-empty-line-before)",
  },
  Object {
    "column": 5,
    "line": 14,
    "rule": "block-opening-brace-space-before",
    "severity": "error",
    "text": "Expected single space before \\"{\\" (block-opening-brace-space-before)",
  },
  Object {
    "column": 15,
    "line": 28,
    "rule": "no-extra-semicolons",
    "severity": "error",
    "text": "Unexpected extra semicolon (no-extra-semicolons)",
  },
  Object {
    "column": 7,
    "line": 31,
    "rule": "number-leading-zero",
    "severity": "error",
    "text": "Expected a leading zero (number-leading-zero)",
  },
]
`;
